neat_ml/opencv/detection.py

```diff
@@ -0,0 +1,252 @@
+from pathlib import Path
+from typing import Any, Dict, Tuple, Sequence
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import skimage.color
+from joblib import Memory
+from tqdm.auto import tqdm
+
+import warnings
+
+__all__: Sequence[str] = [
+    "collect_tiff_paths", 
+    "_detect_single_image",
+    "build_df_from_img_paths", 
+    "_save_debug_overlay",
+    "run_opencv"
+]
+
+def collect_tiff_paths(
+    root: str | Path
+) -> list[str]:
+    """
+    Recursively locate every TIFF image beneath a given directory.
+
+    Parameters
+    ----------
+    root : str or Path
+        Root directory to search for files.
+
+    Returns
+    -------
+    list[str]
+        List of absolute POSIX-style file paths for each .tiff file found.
+    """
+    root_path = Path(root).expanduser().resolve()
+    return [str(Path(p).resolve()) for p in root_path.glob("**/*.tiff")]
```

Comment:
and just so I'm clear--what is it about this implementation that is POSIX-only as the docs describe? it mostly looks portable to me from a quick glance